scipy/stats/morestats.py

```diff
@@ -1241,7 +1262,8 @@ def shapiro(x, a=None, reta=False):
     y = sort(x)
     a, w, pw, ifault = statlib.swilk(y, a[:N//2], init)
     if ifault not in [0, 2]:
-        warnings.warn(str(ifault))
+        warnings.warn("There has been a problem in the computation. The "
```

Comment:
Maybe add ``in `shapiro` ``, so users can more easily locate the issue in their own code.